src/hooks/transactions/index.tsx

```diff
@@ -79,14 +84,19 @@ export function TransactionsUpdater() {
   )
   const onReceipt = useCallback(
     ({ chainId, hash, receipt }) => {
+      if (receipt?.status === 0) {
+        onTxFail?.(new Error('Transaction failed'), receipt)
+      } else {
+        onTxSuccess?.(receipt.transactionHash, receipt)
+      }
```

Comment:
should handle ethereum onTxFail / onTxSuccess in our existing `onReceipt` callback